text/0005-ansible-hostgroup.md

```diff
@@ -0,0 +1,51 @@
+- Feature Name: Allow roles assignment to hostgroup
+- Start Date: 2016-08-04
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+In a similar way foreman_ansible allows assigning roles to hosts we should let users assign roles to host groups. Functionally we want to provide user experience similar to puppet classes in host and hostgroups. There's one difference though: there should be a way of re-ordering the roles or at least the default order must be predictable.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Host groups
+
+Not much alternatives here. We need to implement M:N association between hostgroups and roles and allow for standard hostgroup inheritance.
+
+## Config groups
+
+The Foreman uses config groups for bundling multiple puppet classes together. We could use them for Ansible roles too.
+
+Alternatives of the mapping:
+
+  1. **1:N** - The same approach as with puppet classes. One config group can have multiple roles.
+  1. **1:1** - A role itself is a bundle of Ansible modules which makes it very similar to config group. Therefore it might make more sense to model it as one config group. One of implementations could be using config groups for modeling roles directly (probably via STI).
+  1. **no config groups** - Don't use config groups for roles at all. Roles have similar purpose as config groups, but there's nothing to edit on them in Foreman (we only import them). Therefore we can keep the both models to live side by side.
```

Comment:
1:N or 1:1 (which is the one that makes conceptually most sense) don't make configuration any easier, so I'd say don't link config groups and roles, it just doesn't make sense and makes things more complicated